src/modules/mod/slash-commands/profile-presets.command.ts

```diff
@@ -0,0 +1,307 @@
+      where: {
+        id,
+      },
+    });
+
+    if (!foundPreset)
+      return this.sendError("Can't find preset by ID you provide", arg);
+
+    if (!isValidJson(json))
+      return this.sendError("JSON you provided is not valid", arg);
+
+    // Normalize JSON for storing in db (remove spaces etc.)
+    foundPreset.json = JSON.stringify(JSON.parse(json));
+    await foundPreset.save();
+
+    return this.sendSuccess("Preset updated!", arg);
+  }
+
+  public async deletePreset(arg: CmdArg<CmdType.SLASH_COMMAND>) {
+    const id = arg.options.getString("id", true);
+
+    if (!id.match(/^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$/))
+      return this.sendError("Invalid ID", arg);
+
+    const foundPreset = await ProfilePresetEntity.findOne({
+      where: {
+        id,
+      },
+    });
+
+    if (!foundPreset)
+      return this.sendError("Can't find preset by ID you provide", arg);
+    await foundPreset.remove();
+
+    return this.sendSuccess("Preset deleted!", arg);
+  }
+
+  public async listPresets(arg: CmdArg<CmdType.SLASH_COMMAND>) {
+    const presets = await ProfilePresetEntity.find();
+
+    if (!presets.length)
+      return this.sendError(
+        `Unfortunatelly, we doesn't have any presets in the database`,
+        arg
+      );
+
+    const { list } = getNavigationSetup();
+    const actionRow = new ActionRowBuilder<ButtonBuilder>().addComponents(list);
+    const paginationHelper = new PaginationHelper(presets, {
+      elementsOnPage: 1,
+    });
+
+    const [firstPreset] = paginationHelper.createPage();
+    const componentCollector = arg.channel?.createMessageComponentCollector({
+      componentType: ComponentType.Button,
+      filter: (int) =>
+        int.user.id === arg.user.id && int.message.interaction?.id === arg.id,
+      time: 5 * 1000 * 60000, // 5 minutes
+    });
+
+    if (!componentCollector)
+      return this.sendError("Can't create component collector :(", arg);
+
+    await arg.editReply({
+      content: this.createContent(firstPreset, paginationHelper),
+      embeds: [this.createEmbedFromJson(firstPreset)],
+      components: [actionRow],
+    });
+
+    return componentCollector.on("collect", (int) => {
+      switch (int.customId) {
+        case NavigationButtons.TO_LEFT:
+        case NavigationButtons.TO_RIGHT:
+          const preset = {
+            [NavigationButtons.TO_LEFT]: () =>
+              paginationHelper.prevPage().createPage()[0],
+            [NavigationButtons.TO_RIGHT]: () =>
+              paginationHelper.nextPage().createPage()[0],
+          };
+
+          const getPreset = preset[int.customId]();
+          int.update({
+            content: this.createContent(getPreset, paginationHelper),
+            embeds: [this.createEmbedFromJson(getPreset)],
+          });
+          break;
+        case NavigationButtons.TO_STOP:
+          componentCollector.stop();
+          arg.deleteReply();
+          break;
+      }
```

Comment:
Do we really wanna sell readability for fancy code? I don't think so 🦝

Reply:
Says the one with big ass class of sub-commands. 

But I think that it really isn't a big deal to revert it.

Reply:
You need to destructor big files when you see that some of their parts can be re-used in future and here are place to store smaller ones.

In our case all code in this command logically belongs to it, and we will not use it in any other part of our system. So it's not really worth to slice it for a smaller files just for sake to make it short.

You can move something like DB operation to repositories for example. Can explain to you this later if you want. 

*we probably should do that and maybe write some SQL queries for more speed? Don't need that "efficancy" right now but will be useful in a future :)

You don't need to make all changes that I stated above, just read and think about it. Anyway we will come to this at some point. Rewriting a ton of shit in a future will be cool xD

Reply:
> Rewriting a ton of shit in a future will be cool xD

:despairge:

Will leave it as it is right now. Rewrite later if it scratches the eyes of yours